### PR TITLE
Adds example group-level analysis for phase-coherence effect

### PR DESCRIPTION
**Pull request by JoseAlanis**
- Plot the group-level effect of predictor phase-coherence in LIMO dataset (results consistent with Fig. 4 B3 in [[1]](https://www.hindawi.com/journals/cin/2011/831409/)
- Add 95 % bootstrap CI.
- [WIP]: Bootstrap T-values results.

Rendered fig.:
mean_beta_pc

### CODE REVIEW

**Comment by JoseAlanis** [comment]
sorry, don't pay attention to the commit message (forgot to update from other commit)

**Comment by dengemann** [comment]
ping us when you need feedback.

**Comment by jona-sassenhagen** [comment]
Looks like a decent start.

**Comment by dengemann** [comment]
I notice a double bootstrap here.
One is for the subjects ?
For now I would keep the ERPs fixed. We can later propagate the uncertainty from level 1 to 2. What do you think ?

**Comment by jona-sassenhagen** [comment]
Simpler is better I guess. Reduce to the core added element and reduce overlap with other places.

**Comment by JoseAlanis** [comment]
> I notice a double bootstrap here.
> One is for the subjects ?
> For now I would keep the ERPs fixed. We can later propagate the uncertainty from level 1 to 2. What do you think ?

double bootstrap? sorry I'm not sure what you mean. I'm taking random samples of subjects form the previosly computed betas-array; it contains the lm-results for each subject in th LIMO dataset. The average ERP is just the average of this results and the CI is computed with the random samples of subjects in line 144.

**Comment by dengemann** [comment]
> The average ERP is just the average of this results and the CI is computed with the random samples of subjects in line 144.

You are right, I did not look carefully. You just looped over subjects. The heat in Paris ...

**Comment by JoseAlanis** [comment]
Hey guys, I added the code for the group-level one-sample t-test and bootstrap-t.
I computed the upper and lower boundaries for the bootstrap t-test (see lines 234-265). However, I'm not entirely sure about the computation of the p-values for the bootstrap-t. Essentially they should be equivalent to "the average number of times the T-values obtained from original data are above or below the bootstrap quantiles" (see [[1]](https://www.hindawi.com/journals/cin/2011/831409/)) ... still struggling to get the idea behind this procedure.

Anyways. Here the rendered results from the one-sample t-test (unmasked as no p-values so far) the fig are consistent with Fig. 4 B1 and B2 in [[1]](https://www.hindawi.com/journals/cin/2011/831409/)

1samp_tt
1samp_tt_topo

**Comment by dengemann** [comment]
@JoseAlanis can you show the T histograms along time? They are H1 and should thus be nice to look at, no?

**Comment by JoseAlanis** [comment]
Hey, good idea, something like this?

.120 s
thist_120

.160 s
thist_160

**Comment by JoseAlanis** [comment]
Hey I changed the bootstrap-t part a little bit, I misunderstood the approach the first time but should be consistent with the LIMO-paper now.
For completeness here are the rendered results of the group-level one-sample t-test (histograms are above).

t-image
t-map

**Comment by dengemann** [comment]
The histograms are not centered.
Any idea why that is?

**Comment by JoseAlanis** [comment]
The boot-t was not quite right. I adapted the code to be consistent with whats done in the LIMO-toolbox; i.e., centering the bootstrap samples around the population mean (i.e., mean of original betas) and dividing by the standard error of the bootstrap sample to obtain the sample T-values.

Here the updated results:

Robust CI (computed using boot-t quantiles) for the group-level effect of phase-coherence:
beta_pc

Histogram of boot-t at around .120 s at electrode C1:
boot_t_120ms

Group-level one-sample t-test for phase-coherence showing full electrode*time space with significance level derived using boot-t technique (significant values are above or below boot-t quantiles):
1sampttest_masked

File: examples/group_level/plot_group_level_effects.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+###############################################################################
+
+# at ~ .200 seconds
+plt.hist(boot_t.reshape((boot_t.shape[0],
+                         n_channels,
+                         n_times))[:, pick, time_ind_200], bins=200)
+plt.axvline(x=lower_t[pick, time_ind_200], color='r')
+plt.axvline(x=upper_t[pick, time_ind_200], color='r')
+plt.title('electrode %s, time ~ .200 s' % electrode)
```
Comment (jona-sassenhagen) [inline examples/group_level/plot_group_level_effects.py]: I'd really loop over times here instead. The code is already very long.

**Comment by jona-sassenhagen** [comment]
What are we waiting for here?

**Review by jona-sassenhagen — CHANGES_REQUESTED** [review]

File: examples/group_level/plot_group_level_effects.py (hunk 1 above)
Comment (JoseAlanis) [reply examples/group_level/plot_group_level_effects.py]: the idea was to show that the histograms are centered around zero. I think one time-period is enough to make this point. I'll just delete the other two.

**Pull request merged**